### PR TITLE
Show scenario description on test job infopanel

### PR DESCRIPTION
**Pull request by andrii-suse**
https://progress.opensuse.org/issues/45887

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Users.pm [hunk 1]
```diff
@@ -63,7 +63,7 @@ __PACKAGE__->add_timestamps;
 __PACKAGE__->set_primary_key('id');
 __PACKAGE__->has_many(api_keys => 'OpenQA::Schema::Result::ApiKeys', 'user_id');
 __PACKAGE__->has_many(
-    development_sessions => 'OpenQA::Schema::Result::DevelopmentSessions',
+    development_sessions => 'OpenQA::Schema::Result::DeveloperSessions',
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Users.pm]: I wonder, how is this related? Does it fix something?

File: t/ui/18-tests-details.t [hunk 2]
```diff
@@ -446,6 +448,8 @@ my $post
 $get         = $t->get_ok($baseurl . 'tests/99963')->status_is(200);
 @worker_text = $get->tx->res->dom->find('#assigned-worker')->map('all_text')->each;
 like($worker_text[0], qr/[ \n]*Assigned worker:[ \n]*localhost:1[ \n]*/, 'worker still displayed when job set to done');
+@group_description_text = $get->tx->res->dom->find('#group-description')->map('all_text')->each;
+like($group_description_text[0], qr/[ \n]*##Test description\s+with bugref bsc#1234[ \n]*/, 'test group description is missing');
```
Comment (okurz) [inline t/ui/18-tests-details.t]: Hm, why this repetition?

File: t/ui/18-tests-details.t [hunk 3]
```diff
@@ -158,6 +158,8 @@ my @worker_text = $get->tx->res->dom->find('#assigned-worker')->map('all_text')-
 like($worker_text[0], qr/[ \n]*Assigned worker:[ \n]*localhost:1[ \n]*/, 'worker displayed when job running');
 my @worker_href = $get->tx->res->dom->find('#assigned-worker a')->map(attr => 'href')->each;
 is($worker_href[0], '/admin/workers/1', 'link to worker correct');
+my @group_description_text = $get->tx->res->dom->find('#group-description')->map('all_text')->each;
```
Comment (okurz) [inline t/ui/18-tests-details.t]: I am confused, what are you referring to with "group description text"?

File: docker/openqa/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,8 @@
 FROM registry.opensuse.org/devel/openqa/containers/openqa_dev:latest
 ENV LANG en_US.UTF-8
 
+RUN zypper in -y -C lsof
```
Comment (okurz) [inline docker/openqa/Dockerfile]: that's actually a good idea as a workaround as long as https://github.com/os-autoinst/openQA/pull/1968#issuecomment-456857026 is not solved

@Martchus WDYT?

Comment (Martchus) [reply docker/openqa/Dockerfile]: Yes, but I suppose one would need to add the dependency to `docker/travis_test/Dockerfile`. And I belief only *after* that change has been merged itself the Docker image is rebuilt and used for further PRs.

So don't add the workaround here. And I'd say @coolo should just accept the submit request.

File: lib/OpenQA/Schema/Result/Users.pm (hunk 1 above)
Comment (andrii-suse) [reply lib/OpenQA/Schema/Result/Users.pm]: Yes I saw a warning and decided that it worth to fix it. OK I will make it consistent and will force push as separate commit once all rest is resolved

File: t/ui/18-tests-details.t (hunk 2 above)
Comment (andrii-suse) [reply t/ui/18-tests-details.t]: I tried to do the same way as test for assigned-worker = let's say we check  it second time to be really sure that it isn't gone after refresh. Or I can remove it as well

File: t/ui/18-tests-details.t (hunk 3 above)
Comment (andrii-suse) [reply t/ui/18-tests-details.t]: I tried to do the same way as test for assigned-worker, it uses `woker_text` variable. Test group is referenced as `$job->group` in code and description is `$job->group->description`. Thus variable is group_description + _text .

File: docker/openqa/Dockerfile (hunk 4 above)
Comment (andrii-suse) [reply docker/openqa/Dockerfile]: I wanted to make sure that tests do pass and then either remove workaround from openqa/Dockerfile or maybe keep it just for some time and add lsof dependency into  travis_test/Dockerfile (?)

**Comment by coolo** [comment]
Thanks @andrii-suse for your contribution

File: docker/openqa/Dockerfile (hunk 4 above)
Comment (andrii-suse) [reply docker/openqa/Dockerfile]: It didn't work so I give up trying to inject the fix here

File: templates/test/infopanel.html.ep [hunk 5]
```diff
@@ -96,5 +96,10 @@
                         </span>
                     </div>
                 % }
+                % if ($job && $job->group && $job->group->description && length($job->group->description)) {
+                    <div id="group-description">
+                            %= $job->group->description
```
Comment (okurz) [inline templates/test/infopanel.html.ep]: Well, then I doubt this is what we want. We were looking for the "testsuite description" as in https://github.com/os-autoinst/openQA/pull/1954/files , not the "job group description" :D

Comment (andrii-suse) [reply templates/test/infopanel.html.ep]: Oh sorry :) I've changed group to scenario - please have a look at new commit

**Comment by okurz** [comment]
Please start the commit messages with a capital letter and squash the two commits regarding the scenario description.

Do you have by chance a local instance of openQA available where we would see the new feature working live in action? As an alternative a screenshot would be nice.

**Comment by andrii-suse** [comment]
screenshot from 2019-01-25 10-25-37

^ this is how it may look - description is at the bottom of infopanel ("Simple kde test ...")

File: templates/test/infopanel.html.ep [hunk 6]
```diff
@@ -96,5 +96,10 @@
                         </span>
                     </div>
                 % }
+                % if ($job && $job->scenario && $job->scenario->description && length($job->scenario->description)) {
```
Comment (Martchus) [inline templates/test/infopanel.html.ep]: I would avoid that many method calls just for getting the same value again and again. According to @kraih it is not the best practice in Perl performance-wise. Also the check for `length` is useless because an empty string evaluates to `false` in Perl.

So I'd rather add this line without the multiple method calls and length check to the controller and stash the scenario description there.

File: t/ui/18-tests-details.t [hunk 7]
```diff
@@ -446,6 +448,8 @@ my $post
 $get         = $t->get_ok($baseurl . 'tests/99963')->status_is(200);
 @worker_text = $get->tx->res->dom->find('#assigned-worker')->map('all_text')->each;
 like($worker_text[0], qr/[ \n]*Assigned worker:[ \n]*localhost:1[ \n]*/, 'worker still displayed when job set to done');
+@scenario_description = $get->tx->res->dom->find('#scenario-description')->map('all_text')->each;
+like($scenario_description[0], qr/[ \n]*Simple kde test, before advanced_kde[ \n]*/, 'scenario description is missing');
```
Comment (Martchus) [inline t/ui/18-tests-details.t]: This test isn't any different compared to the one in line 162, is it? And I'm wondering about the test name 'scenario description is missing'. To me it looks that quite the opposite is checked here.

**Comment by andrii-suse** [comment]
> screenshot from 2019-01-25 10-25-37
> 
> ^ this is how it may look - description is at the bottom of infopanel ("Simple kde test ...")

Alternatively, it may look like this (on the same line in header after "Results for" ). Which is better?

screenshot from 2019-01-25 11-32-58

File: templates/test/infopanel.html.ep (hunk 6 above)
Comment (Martchus) [reply templates/test/infopanel.html.ep]: I just saw that we already have a convenience function for the description. So just write it like this:

```
if (my $scenario_description = $job ? $job->scenario_description : undef) {
```

**Comment by okurz** [comment]
> Alternatively, it may look like this (on the same line in header after "Results for" )

Well, the testsuite description is a multi-line field so I doubt we can put it in the header line. Go for the former approach, that should work

**Comment by andrii-suse** [comment]
Thank  you all for the corrections - I hope I covered everything which was discussed in reworked commits - let me know if anything still needs to be done.

File: t/ui/18-tests-details.t (hunk 7 above)
Comment (andrii-suse) [reply t/ui/18-tests-details.t]: Yes, it checks the same after some work - just to make sure that description is still displayed after workflow, but I can remove it if you think it is excessive.

File: templates/test/infopanel.html.ep (hunk 6 above)
Comment (Martchus) [reply templates/test/infopanel.html.ep]: The thing I initially didn't like and still think should be improved was the unnecessarily repeated method call. It should be `if (my $scenario_description = $job->scenario_description) {` to use the variable inside the block. (That saves more than one method call because `$job->scenario_description` does further method calls internally.)

**Comment by okurz** [comment]
The checks have failed in https://travis-ci.org/os-autoinst/openQA/jobs/485385751#L963 due to what I think is fixed meanwhile in the master branch. Could you please rebase your branch the latest master branch? This will include the fix and retrigger the checks.

**Comment by andrii-suse** [comment]
> The checks have failed in https://travis-ci.org/os-autoinst/openQA/jobs/485385751#L963 due to what I think is fixed meanwhile in the master branch.

I did pull in the morning before my recent commits, so all upstream commits are in the branch. 
https://github.com/andrii-suse/openQA/commits/test_infopanel_add_group_description

I will try to close/reopen the pull request to trigger the build.
